src/main/java/com/esri/samples/map/generate_offline_map_overrides/GenerateOfflineMapOverridesController.java

```diff
@@ -0,0 +1,282 @@
+import com.esri.arcgisruntime.geometry.Envelope;
+import com.esri.arcgisruntime.geometry.GeometryEngine;
+import com.esri.arcgisruntime.geometry.Point;
+import com.esri.arcgisruntime.layers.FeatureLayer;
+import com.esri.arcgisruntime.layers.Layer;
+import com.esri.arcgisruntime.loadable.LoadStatus;
+import com.esri.arcgisruntime.mapping.ArcGISMap;
+import com.esri.arcgisruntime.mapping.view.Graphic;
+import com.esri.arcgisruntime.mapping.view.GraphicsOverlay;
+import com.esri.arcgisruntime.mapping.view.MapView;
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+import com.esri.arcgisruntime.security.AuthenticationManager;
+import com.esri.arcgisruntime.security.DefaultAuthenticationChallengeHandler;
+import com.esri.arcgisruntime.symbology.SimpleLineSymbol;
+import com.esri.arcgisruntime.tasks.geodatabase.GenerateGeodatabaseParameters;
+import com.esri.arcgisruntime.tasks.geodatabase.GenerateLayerOption;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapJob;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapParameterOverrides;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapParameters;
+import com.esri.arcgisruntime.tasks.offlinemap.GenerateOfflineMapResult;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapParametersKey;
+import com.esri.arcgisruntime.tasks.offlinemap.OfflineMapTask;
+import com.esri.arcgisruntime.tasks.tilecache.ExportTileCacheParameters;
+
+public class GenerateOfflineMapOverridesController {
+
+  @FXML private MapView mapView;
+  @FXML private Spinner<Integer> minScaleLevelSpinner;
+  @FXML private Spinner<Integer> maxScaleLevelSpinner;
+  @FXML private Spinner<Integer> extentBufferDistanceSpinner;
+  @FXML private Spinner<Integer> minHydrantFlowRateSpinner;
+  @FXML private CheckBox systemValvesCheckBox;
+  @FXML private CheckBox serviceConnectionsCheckBox;
+  @FXML private CheckBox waterPipesCheckBox;
+  @FXML private Button generateOfflineMapButton;
+  @FXML private Button cancelJobButton;
+  @FXML private ProgressBar progressBar;
+
+  private ArcGISMap map;
+  private GraphicsOverlay graphicsOverlay;
+  private Graphic downloadArea;
+  private GenerateOfflineMapJob job;
+
+  @FXML
+  private void initialize() {
+    // handle authentication with the portal
+    AuthenticationManager.setAuthenticationChallengeHandler(new DefaultAuthenticationChallengeHandler());
+
+    // create a portal item with the itemId of the web map
+    Portal portal = new Portal("https://www.arcgis.com", true);
+    PortalItem portalItem = new PortalItem(portal, "acc027394bc84c2fb04d1ed317aac674");
+
+    // create a graphics overlay for display the download area
+    graphicsOverlay = new GraphicsOverlay();
+    mapView.getGraphicsOverlays().add(graphicsOverlay);
+
+    // show a red border around the download area
+    downloadArea = new Graphic();
+    graphicsOverlay.getGraphics().add(downloadArea);
```

Comment:
The download area graphic doesn't always draw immediately after user logs in: leading to error when clicking on generate offline map button. If user pans the map slightly the graphic draws fine though.